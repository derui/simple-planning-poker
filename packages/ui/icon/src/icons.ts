/**
 * Icons to be able to use with Icon component
 */
export enum Icons {
  allowBackUp = "arrow-back-up",
  check = "check",
  chevronDown = "chevron-down",
  chevronLeft = "chevron-left",
  chevronRight = "chevron-right",
  circleArrowLeft = "circle-arrow-left",
  clipboardCheck = "clipboard-check",
  clipboard = "clipboard",
  clockHour1 = "clock-hour-1",
  clockHour2 = "clock-hour-2",
  clockHour3 = "clock-hour-3",
  clockHour4 = "clock-hour-4",
  clockHour5 = "clock-hour-5",
  clockHour6 = "clock-hour-6",
  clockHour7 = "clock-hour-7",
  clockHour8 = "clock-hour-8",
  clockHour9 = "clock-hour-9",
  clockHour10 = "clock-hour-10",
  clockHour11 = "clock-hour-11",
  clockHour12 = "clock-hour-12",
  doorExit = "door-exit",
  equal = "equal",
  eye = "eye",
  loader2 = "loader-2",
  main = "mail",
  pencil = "pencil",
  settings = "settings",
  userX = "user-x",
  user = "user",
  users = "users",
  x = "x",
}
