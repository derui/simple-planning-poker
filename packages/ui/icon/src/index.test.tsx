import { test, afterEach, expect } from "vitest";
import { render, cleanup } from "@testing-library/react";
import { Icons } from "./icons";
import { Icon } from "./index";

afterEach(cleanup);

test("should be able to render", async () => {
  const ret = render(<Icon type={Icons.allowBackUp} />);

  expect(ret.container).toMatchSnapshot();
});

test("render other size small", async () => {
  const ret = render(<Icon size="s" type={Icons.check} />);

  expect(ret.container).toMatchSnapshot();
});

test("render size large", () => {
  const ret = render(<Icon size="l" type={Icons.chevronDown} />);

  expect(ret.container).toMatchSnapshot();
});
