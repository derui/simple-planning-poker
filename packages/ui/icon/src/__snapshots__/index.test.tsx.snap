// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render other size small 1`] = `
<div>
  <span
    class="before:inline-block before:flex-none before:[mask-image:url("/static/svg/tabler-icons/check.svg")] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-5 before:h-5 w-5 h-5 flex"
  />
</div>
`;

exports[`render size large 1`] = `
<div>
  <span
    class="before:inline-block before:flex-none before:[mask-image:url("/static/svg/tabler-icons/chevron-down.svg")] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-7 before:h-7 w-7 h-7 flex"
  />
</div>
`;

exports[`should be able to render 1`] = `
<div>
  <span
    class="before:inline-block before:flex-none before:[mask-image:url("/static/svg/tabler-icons/arrow-back-up.svg")] before:[mask-size:cover] before:[mask-position:center] before:[mask-repeat:no-repeat] before:w-6 before:h-6 w-6 h-6 flex"
  />
</div>
`;
